lib/utils.js

```diff
@@ -479,4 +485,61 @@ var buildRequestOptionsForRefreshSAS = function buildRequestOptionsForRefreshSAS
   });
 }
 
-exports.buildRequestOptionsForRefreshSAS = buildRequestOptionsForRefreshSAS;
\ No newline at end of file
+exports.buildRequestOptionsForRefreshSAS = buildRequestOptionsForRefreshSAS;
+
+/**
+ * @param {object} conditionalHeaders - Conditional headers on the following form
+ * ```js
+ * {
+ *    ifModifiedSince: new Date(),      // Specify this to perform the operation only if the resource has been modified since the specified time. (optional)
+ *    ifUnmodifiedSince: new Date(),    // Specify this to perform the operation only if the resource has not been modified since the specified date/time. (optional)
+ *    ifMatch: '...',                   // ETag value. Specify this to perform the operation only if the resource's ETag matches the value specified. (optional)
+ *    ifNoneMatch: '...'                // ETag value. Specify this to perform the operation only if the resource's ETag does not match the value specified. (optional)
+ * }
+ *```
+ */
+var setConditionalHeaders = function setConditionalHeaders(headers, conditionalHeaders) {
+  if (conditionalHeaders) {
+    if (conditionalHeaders.ifModifiedSince){
+      assert(conditionalHeaders.ifModifiedSince instanceof Date,
+        'If specified, the `options.ifModifiedSince` must be a Date');
+      headers['if-modified-since'] = conditionalHeaders.ifModifiedSince.toUTCString();
+    }
+    if (conditionalHeaders.ifUnmodifiedSince) {
+      assert(conditionalHeaders.ifUnmodifiedSince instanceof Date,
+        'If specified, the `options.ifUnmodifiedSince` must be a Date');
+      headers['if-unmodified-since'] = conditionalHeaders.ifUnmodifiedSince.toUTCString();
+    }
+    if (conditionalHeaders.ifMatch) headers['if-match'] = conditionalHeaders.ifMatch;
+    if (conditionalHeaders.ifNoneMatch) headers['if-none-match'] = conditionalHeaders.ifNoneMatch;
+  }
+}
+
+exports.setConditionalHeaders = setConditionalHeaders;
+
+/**
+ * Computes a signature for the specified string using the HMAC-SHA256 algorithm.
+ *
+ * @param {string} accessKey - The access key
+ * @param {string} stringToSign - The UTF-8-encoded string to sign.
+ * @return A String that contains the HMAC-SHA256-encoded signature.
+ */
+var getHmacSha256Sign = function getHmacSha256Sign(accessKey, stringToSign) {
```

Comment:
Let's rename this to `hmacSha256` there is certainly no `get` involved, one might be able to say `computeHmacSha256`... But I would be very happy just calling it `hmacSha256`, short and concise.